### PR TITLE
mgr/dashboard: Reduce code duplication through TableActionComponent UnitTests

### PR DESCRIPTION
**Pull request by p-na**
Fixes: http://tracker.ceph.com/issues/40399

Signed-off-by: Patrick Nawracay <pnawracay@suse.com>

- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by p-na** [comment]
Please note that this PR does not yet contain the suggested change to the unique title that is composed of the permissions set:

Example: `true-true-false`
Proposal: `allowCreate-allowUpdate-denyDelete`

If the proposal is accepted, I will be glad to adapt the code! Of course I'm all ears for other proposals!

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-list/iscsi-target-list.component.spec.ts [hunk 1]
```diff
@@ -173,158 +173,46 @@ describe('IscsiTargetListComponent', () => {
...
-        expect(tableActions.tableActions.length).toBe(0);
-        expect(tableActions.tableActions).toEqual([]);
-      });
+  it('should test all TableActions combinations', () => {
+    const permissionHelper: PermissionHelper = new PermissionHelper(component.permission);
+    const tableActions: TableActionsComponent = permissionHelper.setPermissionsAndGetActions(
+      component.tableActions
+    );
+
+    console.log(tableActions);
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-list/iscsi-target-list.component.spec.ts]:
```suggestion
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-snapshot-list/rbd-snapshot-list.component.spec.ts [hunk 2]
```diff
@@ -212,209 +212,85 @@ describe('RbdSnapshotListComponent', () => {
...
-          component.tableActions[6]
-        ]);
-      });
+  it('should test all TableActions combinations', () => {
+    const permissionHelper: PermissionHelper = new PermissionHelper(component.permission);
+    const tableActions: TableActionsComponent = permissionHelper.setPermissionsAndGetActions(
+      component.tableActions
+    );
+
+    console.log(tableActions);
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-snapshot-list/rbd-snapshot-list.component.spec.ts]:
```suggestion
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.spec.ts [hunk 3]
```diff
@@ -169,160 +169,46 @@ describe('NfsListComponent', () => {
...
-        expect(tableActions.tableActions.length).toBe(0);
-        expect(tableActions.tableActions).toEqual([]);
-      });
+  it('should test all TableActions combinations', () => {
+    const permissionHelper: PermissionHelper = new PermissionHelper(component.permission);
+    const tableActions: TableActionsComponent = permissionHelper.setPermissionsAndGetActions(
+      component.tableActions
+    );
+
+    console.log(tableActions);
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-list/nfs-list.component.spec.ts]:
```suggestion
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/mgr-modules/mgr-module-list/mgr-module-list.component.spec.ts [hunk 4]
```diff
@@ -50,57 +50,46 @@ describe('MgrModuleListComponent', () => {
...
-        expect(tableActions.tableActions.length).toBe(0);
-        expect(tableActions.tableActions).toEqual([]);
-      });
+  it('should test all TableActions combinations', () => {
+    const permissionHelper: PermissionHelper = new PermissionHelper(component.permission);
+    const tableActions: TableActionsComponent = permissionHelper.setPermissionsAndGetActions(
+      component.tableActions
+    );
+
+    console.log(tableActions);
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/mgr-modules/mgr-module-list/mgr-module-list.component.spec.ts]:
```suggestion
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-list/rgw-bucket-list.component.spec.ts [hunk 5]
```diff
@@ -43,160 +43,46 @@ describe('RgwBucketListComponent', () => {
...
-        expect(tableActions.tableActions.length).toBe(0);
-        expect(tableActions.tableActions).toEqual([]);
-      });
+  it('should test all TableActions combinations', () => {
+    const permissionHelper: PermissionHelper = new PermissionHelper(component.permission);
+    const tableActions: TableActionsComponent = permissionHelper.setPermissionsAndGetActions(
+      component.tableActions
+    );
+
+    console.log(tableActions);
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-bucket-list/rgw-bucket-list.component.spec.ts]:
```suggestion
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-list/rgw-user-list.component.spec.ts [hunk 6]
```diff
@@ -37,160 +36,46 @@ describe('RgwUserListComponent', () => {
...
-        expect(tableActions.tableActions.length).toBe(0);
-        expect(tableActions.tableActions).toEqual([]);
-      });
+  it('should test all TableActions combinations', () => {
+    const permissionHelper: PermissionHelper = new PermissionHelper(component.permission);
+    const tableActions: TableActionsComponent = permissionHelper.setPermissionsAndGetActions(
+      component.tableActions
+    );
+
+    console.log(tableActions);
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-list/rgw-user-list.component.spec.ts]:
```suggestion
```

File: src/pybind/mgr/dashboard/frontend/src/app/core/auth/role-list/role-list.component.spec.ts [hunk 7]
```diff
@@ -44,160 +44,46 @@ describe('RoleListComponent', () => {
...
-        expect(tableActions.tableActions.length).toBe(0);
-        expect(tableActions.tableActions).toEqual([]);
-      });
+  it('should test all TableActions combinations', () => {
+    const permissionHelper: PermissionHelper = new PermissionHelper(component.permission);
+    const tableActions: TableActionsComponent = permissionHelper.setPermissionsAndGetActions(
+      component.tableActions
+    );
+
+    console.log(tableActions);
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/core/auth/role-list/role-list.component.spec.ts]: Same for remaining occurences :)

```suggestion
```

File: src/pybind/mgr/dashboard/frontend/src/testing/unit-test-helper.ts [hunk 8]
```diff
@@ -40,64 +41,62 @@ export function configureTestBed(configuration, useOldMethod?) {
...
+          this.permission.update = update;
+          this.permission.delete = deleteP;
+
+          this.tac = new TableActionsComponent();
+          this.tac.selection = new CdTableSelection();
+          this.tac.tableActions = [...tableActions];
+          this.tac.permission = this.permission;
+          this.tac.ngOnInit();
+
+          result[create + '-' + update + '-' + deleteP] = {
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/testing/unit-test-helper.ts]: One possible solution would be to write only the name of the enabled permissions.
p.e. : 'create' and 'create-update' 
Its shorter and also easy to know what case was tested.
We could also user a ',' to separate the permissions.

Comment (p-na) [reply src/pybind/mgr/dashboard/frontend/src/testing/unit-test-helper.ts]: Adapted it as requested (`create,update,delete`). For the tests which don't have any permissions set, it'll create the text `no-permissions`.

**Comment by p-na** [comment]
jenkins test make check

**Comment by p-na** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins retest this please

**Comment by p-na** [comment]
jenkins retest this please

**Pull request merged**